newIDE/app/src/Profile/AuthenticatedUserProvider.js

```diff
@@ -649,48 +671,32 @@ export default class AuthenticatedUserProvider extends React.Component<
     }
   };
 
-  _notifyUserAboutEmailVerificationAndAdditionalInfo = () => {
-    const { profile } = this.state.authenticatedUser;
+  _notifyUserAboutEmailVerification = () => {
+    const { profile, firebaseUser } = this.state.authenticatedUser;
     if (!profile) return;
+    if (firebaseUser && firebaseUser.emailVerified) return;
+
+    const now = Date.now();
     // If the user has not verified their email when logging in we show a dialog to do so.
     // - If they just registered, we don't send the email again as it will be sent automatically,
     // nor do we show a button to send again.
     // - If they are just logging in, we don't send the email but we show a button to send again.
     // Use a boolean to show the dialog only once.
-    const accountAgeInMs = Date.now() - profile.createdAt;
-    const hasJustCreatedAccount = accountAgeInMs < 1000 * 10; // 10 seconds.
-    if (
-      this.state.authenticatedUser.firebaseUser &&
-      !this.state.authenticatedUser.firebaseUser.emailVerified &&
-      !this._hasNotifiedUserAboutEmailVerification
-    ) {
-      setTimeout(
-        () =>
-          this.openEmailVerificationDialog({
-            open: true,
-            sendEmailAutomatically: false,
-            showSendEmailButton: !hasJustCreatedAccount,
-          }),
-        1000
-      );
-    } else {
-      // If the user has not filled additional info, we show a dialog to do so.
-      this._notifyUserAboutAdditionalInfo();
-    }
-  };
-
-  _notifyUserAboutAdditionalInfo = () => {
-    const profile = this.state.authenticatedUser.profile;
-    if (!profile) return;
-    // If the user has not filled their additional information, show
-    // the dialog to fill it, but ensure they have closed the email verification dialog first.
-    // Use a boolean to show the dialog only once.
-    if (
-      profile &&
-      !this._hasNotifiedUserAboutAdditionalInfo &&
-      shouldAskForAdditionalUserInfo(profile)
-    ) {
-      setTimeout(() => this.openAdditionalUserInfoDialog(true), 1000);
+    const accountAgeInMs = now - profile.createdAt;
+    const hasJustCreatedAccount = accountAgeInMs < TEN_SECONDS;
+    const hasAlreadyBeenDisplayedInTheLast30Minutes = this
+      ._lastEmailVerificationDialogDisplayDate
+      ? now - this._lastEmailVerificationDialogDisplayDate < THIRTY_MINUTES
+      : false;
+    if (!hasAlreadyBeenDisplayedInTheLast30Minutes) {
```

Comment:
I think this logic is too complex, we have 2 booleans now.
The reason this is shown too many times is because we've removed `!this._hasNotifiedUserAboutEmailVerification` from the condition. (which allowed to show this dialog only once before, as it was reset on every launch)